src/aws.ts

```diff
@@ -687,16 +688,8 @@ function getTableArgument(op: string, args: Argument[] | Expr[]) {
     );
   }
 
-  const tableRef = tableProp.expr;
+  const table = tryFindIntegration(tableProp.expr);
 
-  if (!isReferenceExpr(tableRef)) {
-    throw new SynthError(
-      ErrorCodes.Expected_an_object_literal,
-      `First argument into ${op} should be an input with a property 'Table' that is a Table.`
-    );
-  }
-
-  const table = tableRef.ref();
```

Comment:
Why don't you just use `evalToConstant`? and then check if it is defined and a `Table`.

Reply:
I really don't like `tryFindIntegration`, its better to evaluate than to search. 

For example they may do something like `table.tableArn` and you'd end up returning the `table` part, not the `targetArn` part, which should fail.

Reply:
`evalToCostant` doesn't return a `Table`.

Reply:
hmm, I think you are right, we filter out only primitives.

Does `tryFind` search or does it evaluate down to a free variable that is an integration?

Reply:
It does evaluate, following reference expressions and applying expressions to extract the actual value. Maybe poorly named but built for this purpose. We can improve later, this is an easy unblocking fix.

Reply:
I hope you are right because of how we use it... but this block concerns me

https://github.com/functionless/functionless/blob/main/src/util.ts#L289

Reply:
That's evalToConstant?